main.go

```diff
@@ -535,9 +548,21 @@ func (c *controller) saveHashring(hashring []receive.HashringConfig) error {
 		return err
 	}
 
+	c.configmapLastSuccessfulChangeTime.Set(float64(time.Now().Unix()))
+	c.configmapHash.Set(hashAsMetricValue(buf))
 	return nil
 }
 
+// hashAsMetricValue generates metric value from hash of data.
+func hashAsMetricValue(data []byte) float64 {
```

Comment:
This function already implemented in [Thanos](https://github.com/thanos-io/thanos/pull/1378) as well. It could be exported in there and used in here.

Reply:
No need to import I think.